### DISCUSSION TITLE
Why I need `$file.name` in the first example and must use `$file` in the second example?

### DISCUSSION THREAD

**Discussion by danli349** [nushell/nushell#11397] [Q&A] [answered]
Hello

When I transform the `svg` files in a folder to `pdf` files, I can use 2 different methods: 
1st:
```bash
for file in (ls *.svg) {`C:\Users\lidan\Inkscape\bin\inkscape.exe` 
[ $file.name --export-area-drawing  --batch-process  --export-type pdf --export-filename $file.name]
}
```

2nd:
```bash
^ls > files
open files | lines | par-each { |file| `C:\Users\lidan\Inkscape\bin\inkscape.exe` 
[ $file --export-area-drawing  --batch-process  --export-type pdf --export-filename $file]
}
```

Why I must `$file.name` in the first example and must use `$file` in the second example?

**Comment by fdncred** [comment, marked as answer]
In the 1st example, you're using the built in `ls` command which returns structured data. In the 2nd example, you're using the operating system's `ls` command which does not return structured data.

To see the difference between them, in nushell in your terminal do `ls *.svg` and then do `^ls`. You'll see that these are very different outputs.

**Reply by danli349** [reply]
How can save the structured data from the output of the built-in `ls` to a file? Thanks

**Reply by fdncred** [reply]
This is one way `ls | $in.name o>> file`. Here's another `ls | $in.name | save --append foo`. These will save only the file names to a file.

**Reply by danli349** [reply]
@fdncred 
The first way doesn't work
```bash
ls | $in.name o>> file                                          1 12/22/2023 11:32:10 AMError: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[entry #2:1:1]
 1 │ ls | $in.name o>> file
   ·               ─┬─
   ·                ╰── expected operator
   ╰────
```
the 2nd way works

**Reply by fdncred** [reply]
In fact, it does.
```nushell
❯ ls | $in.name o>> file
❯ cat file
CODE_OF_CONDUCT.md
CONTRIBUTING.md
Cargo.lock
Cargo.toml
```
You probably have an old version of nushell.

**Reply by danli349** [reply]
Yes, I upgraded and it works.
